### PR TITLE
Add README with examples

### PR DESCRIPTION
**Pull request by Yeti-or**
@zxqfox

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# gulp-eval
+[![NPM version](http://img.shields.io/npm/v/gulp-eval.svg?style=flat)](http://www.npmjs.org/package/gulp-eval) [![Build Status](http://img.shields.io/travis/gulp-bem/gulp-eval/master.svg?style=flat&label=tests)](https://travis-ci.org/gulp-bem/gulp-eval) [![Coverage Status](https://img.shields.io/coveralls/gulp-bem/gulp-eval.svg?branch=master&style=flat)](https://coveralls.io/r/gulp-bem/gulp-eval) [![Dependency Status](http://img.shields.io/david/gulp-bem/gulp-eval.svg?style=flat)](https://david-dm.org/gulp-bem/gulp-eval)
```
Comment (tadatuta) [inline README.md]: @miripiruni found out that's having badges right after title is bad for suggest on npm website. It's better to switch them with description.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+Require commonJS modules, json files or eval expression inside gulp tasks with ease.
```
Comment (tadatuta) [inline README.md]: JSON

File: README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+gulp-eval fills `data` property of [Vinyl](https://github.com/gulpjs/vinyl) files.
+
+## Example of use:
```
Comment (tadatuta) [inline README.md]: Usage example?

File: .travis.yml [hunk 4]
```diff
@@ -4,8 +4,9 @@ language: node_js
 
 matrix:
   include:
-    - node_js: "4.2.2"
+    - node_js: "4.4.3"
+    - node_js: "5.10.1"
```
Comment (qfox) [inline .travis.yml]: Давай всё-таки версию не указывать явно? Достаточно просто 4 и 5

File: README.md (hunk 1 above)
Comment (zxqfox) [inline README.md]: 👍

File: README.md (hunk 2 above)
Comment (qfox) [inline README.md]: CommonJS

Comment (qfox) [inline README.md]: Require CommonJS modules, JSON files or files with JS-expression ...

File: README.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
+# gulp-eval
+[![NPM version](http://img.shields.io/npm/v/gulp-eval.svg?style=flat)](http://www.npmjs.org/package/gulp-eval) [![Build Status](http://img.shields.io/travis/gulp-bem/gulp-eval/master.svg?style=flat&label=tests)](https://travis-ci.org/gulp-bem/gulp-eval) [![Coverage Status](https://img.shields.io/coveralls/gulp-bem/gulp-eval.svg?branch=master&style=flat)](https://coveralls.io/r/gulp-bem/gulp-eval) [![Dependency Status](http://img.shields.io/david/gulp-bem/gulp-eval.svg?style=flat)](https://david-dm.org/gulp-bem/gulp-eval)
+
+Require commonJS modules, json files or eval expression inside gulp tasks with ease.
+
+gulp-eval fills `data` property of [Vinyl](https://github.com/gulpjs/vinyl) files.
```
Comment (qfox) [inline README.md]: Wrap in ticks

Comment (qfox) [inline README.md]: Actually, they are File- (or Vinyl) objects, not files.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+## Example of use:
+```js
+var gulp = require('gulp');
+var concat = require('gulp-concat');
+var gulpEval = require('gulp-eval');
+var through = require('through2');
+
+gulp.task('default', () => {
+  return gulp.src('node_modules/*/package.json')
```
Comment (qfox) [inline README.md]: return is redundant if we using arrow function with one statement

File: README.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    .pipe(gulpEval())
+    .pipe(through.obj(function(file, enc, next) {
+      var data = file.data; // Here we have evaluated package.json
+      file.contents = new Buffer(data.name + '#' + data.version);
+      next(null, file);
```
Comment (qfox) [inline README.md:21]:
``` js
    .pipe(through.obj(function(file, enc, next) {
      var d = file.data; // Here we have evaluated package.json
      if (!d) return next(null);
      next(null, d.name + '#' + d.version);
    })
    .pipe(process.stdout);
```

?

File: README.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    .pipe(concat('all.txt'))
+    .pipe(gulp.dest('dest/'));
+});
+```
+
+## Eval
+You can use eval function without gulp
+
+```js
+var saveEval = require('gulp-eval').eval;
```
Comment (qfox) [inline README.md]: sa**f**e

File: index.js [hunk 9]
```diff
@@ -68,6 +84,10 @@ function wrap(body, extKeys) {
     return wrapper[0] + extKeys + wrapper[1] + body + wrapper[2];
 }
 
+/**
+ * Execute function inside try-catch
+ * code inside try-catch is not optimized so we made this helper
```
Comment (qfox) [inline index.js]: Function's body **with** try-catch won't be optimized, not only inside it

File: README.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    };
+`;
+
+var block = saveEval(requireContent);
+
+console.log(block.elem);
+```
+
+## Context
+You can provide some like-a-global variables into eval
```
Comment (qfox) [inline README.md]: That's cool, but the first purpose of the project is using inside gulp pipes, do we have an option to pass context?

File: README.md [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+## Example of use:
+```js
+var gulp = require('gulp');
+var concat = require('gulp-concat');
+var gulpEval = require('gulp-eval');
+var through = require('through2');
+
+gulp.task('default', () => {
+  return gulp.src('node_modules/*/package.json')
+    .pipe(gulpEval())
```
Comment (qfox) [inline README.md:17]: need an example with passing context

File: README.md [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+    .pipe(through.obj(function(file, enc, next) {
+      var data = file.data; // Here we have evaluated package.json
+      file.contents = new Buffer(data.name + '#' + data.version);
+      next(null, file);
+    }))
+    .pipe(concat('all.txt'))
```
Comment (qfox) [inline README.md]: If you'll decide to left this concat — please use another name. Maybe something like `packages.txt`.

File: index.js [hunk 13]
```diff
@@ -54,6 +62,14 @@ function _eval(content, filename, context) {
 
 module.exports.eval = _eval;
 
+/**
+ * Wrap code with function expression
+ * Use nodejs style default wrapper
+ *
+ * @param {String} body
+ * @param {Array} [extKeys] keys to extend function args
```
Comment (qfox) [inline index.js]: `{string[]}` ?

File: index.js [hunk 14]
```diff
@@ -11,6 +11,14 @@ module.exports = function(context) {
     });
 };
 
+/**
+ * Eval expressions, JSON files or require commonJS modules
+ *
+ * @param {String} content
+ * @param {String} [filename] path to file which content we execute
+ * @param {Object} [context] objects to provide into execute method
+ * @return {Object}
```
Comment (qfox) [inline index.js]: `@returns {*}`

File: README.md [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+
+```js
+var saveEval = require('gulp-eval').eval;
+
+var secretKey = '^___^';
+
+var content =
+`
+    module.exports = {
+        elem: secretKey,
```
Comment (qfox) [inline README.md]: don't need this object with elem for example

File: README.md [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+    };
+`;
+
+var block = saveEval(content, 'file.js', {secretKey: secretKey});
```
Comment (qfox) [inline README.md]:
``` js
var result = safeEval(...);
console.log(result); // ^___^
``
```

**Comment by qfox** [comment]
Thanks, @tadatuta ;-)

@Yeti-or I'm done.

**Pull request merged**